### ISSUE TITLE
Support override insert

### ISSUE THREAD

**Issue by matko** [terminusdb/terminusdb#705]
We want a way to insert data whose ids may or may not already exist. If the id exists, the operation should be a replace. If it doesn't, it is a normal insert.

For example, if we have objects called foo and bar in the database, and someone submits documents with ids bar and baz, bar should be replaced, and baz should be newly inserted.

I propose we add a new flag to the document interface on POST, `override=true`, which toggles this behavior.

**Comment by GavinMendelGleason** [comment]
According to RFC 2616, § 9.6, PUT is used to create or replace a resource:

```
The PUT method requests that the enclosed entity be stored under the supplied Request-URI. If the Request-URI refers to an already existing resource, the enclosed entity SHOULD be considered as a modified version of the one residing on the origin server. If the Request-URI does not point to an existing resource, and that URI is capable of being defined as a new resource by the requesting user agent, the origin server can create the resource with that URI.
```

I think the flag should be "create", a boolean, which if false, will error when the resource does not exist.

**Comment by GavinMendelGleason** [comment]
Fixed in PR #750

**Comment by spl** [comment]
Closed by #750.
